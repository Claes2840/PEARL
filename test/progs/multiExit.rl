init: entry
      if dyn goto fst else snd

fst: from init
     elim ^= 'a
     goto stop

snd: from init
     elim ^= 'b
     goto stop

stop: fi dyn from fst else snd
      exit