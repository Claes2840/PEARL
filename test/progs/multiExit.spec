// None needed
elim1 = 'nil