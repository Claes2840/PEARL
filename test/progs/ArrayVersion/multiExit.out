init_elim_0:
	entry
	if (dyn = 0)
		goto fst_elim_0
		else snd_elim_0


fst_elim_0:
	from init_elim_0
	goto stop_elim_1


snd_elim_0:
	from init_elim_0
	goto stop_elim_2


stop_elim_1:
	fi (dyn = 0)
		from fst_elim_0
		else snd_NULL
	elim += 1
	exit_control_var += 0
	goto exit_merge_0_1


stop_elim_2:
	fi (dyn = 0)
		from fst_NULL
		else snd_elim_0
	elim += 2
	exit_control_var += 1
	goto exit_merge_0_1


exit_merge_0_1:
	fi (exit_control_var < 1)
		from stop_elim_1
		else stop_elim_2
	exit